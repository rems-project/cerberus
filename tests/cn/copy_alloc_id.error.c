

void f (int *p)
{
  unsigned long long p_int = (unsigned long long) p;
  int* q = __cerbvar_copy_alloc_id(p_int + 0ULL, p);
  /*@ assert (ptr_eq(p, q)); @*/
}

int main(void)
{
  int p[1] = {1};
  f(p);
}
