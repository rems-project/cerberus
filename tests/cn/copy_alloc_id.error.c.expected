return code: 1
[1/2]: f -- fail
[2/2]: main -- pass
tests/cn/copy_alloc_id.error.c:4:12: error: Pointer `p` needs allocation ID
  int* q = __cerbvar_copy_alloc_id(p_int + 0ULL, p);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
(UB missing short message): UB_CERB004_unspecified__copy_alloc_id
State file: file:///tmp/state__copy_alloc_id.error.c__f.html
